### PR TITLE
doc: improve package doc and add a note about unbounded task results

### PR DESCRIPTION
**Pull request by rolinh**

### CODE REVIEW

File: workerpool.go [hunk 1]
```diff
@@ -13,7 +13,19 @@
...
+// When the limit of concurrently running workers is reached, submitting a task
+// blocks until a worker is able to pick it up. This behavior is intentional as
+// it prevents from accumulating tasks which could grow unbounded. Therefore,
+// it is the responsibility of the caller to queue up tasks if that's the
+// intended behavior.
+//
+// One caveat is that while the number of concurrently running workers is
+// limited, task results are not and they accumulate until they are collected.
+// Therefore, if a large number of tasks can be expected, the workerpool should
+// be periodically drained (e.g. every 10k tasks).package workerpool
```
Comment (tklauser) [inline workerpool.go:28]:
```suggestion
// be periodically drained (e.g. every 10k tasks).
```

**Review by tklauser — APPROVED** [review]
Thanks for documenting this. LGTM with one minor change (probably an accidental paste).

File: workerpool.go (hunk 1 above)
Comment (rolinh) [reply workerpool.go]: Hah, I fat fingered this with vim. I was wonder where the `package` line went :laughing: . Thanks.

File: README.md [hunk 2]
```diff
@@ -5,7 +5,19 @@
 [![Go Report Card](https://goreportcard.com/badge/github.com/cilium/workerpool)](https://goreportcard.com/report/github.com/cilium/workerpool)
 
 Package workerpool implements a concurrency limiting worker pool. Worker
-routines are spawned on demand as tasks are submitted.
+routines are spawned on demand as tasks are submitted; up to the configured
+limit of concurrent workers.
+
+When the limit of concurrently running workers is reached, submitting a task
+blocks until a worker is able to pick it up. This behavior is intentional as it
+prevents from accumulating tasks which could grow unbounded. Therefore, it is
```
Comment (kaworu) [inline README.md:13]: Should we talk about unbounded goroutine spawned rather than "acummulating tasks"`?

File: README.md [hunk 3]
```diff
@@ -5,7 +5,19 @@
...
+the responsibility of the caller to queue up tasks if that's the intended
+behavior.
+
+One caveat is that while the number of concurrently running workers is limited,
+task results are not and they accumulate until they are collected. Therefore,
+if a large number of tasks can be expected, the workerpool should be
+periodically drained (e.g. every 10k tasks).
```
Comment (kaworu) [inline README.md:20]:
```suggestion
periodically drained (e.g. every ten thousand tasks).
```

**Review by kaworu — COMMENTED** [review]
LGTM, ditto remarks for the `workerpool.go` top-level comment section.

File: README.md (hunk 2 above)
Comment (rolinh) [reply README.md:13]: I don't think so. I really mean "unbounded task list" here, not unbounded pool of goroutines.

**Review by kaworu — APPROVED** [review]

**Pull request merged**